pkg/operator/controller.go

```diff
@@ -84,7 +87,8 @@ func NewController() (*Controller, error) {
 		cvowrapper:     operatorstatus.NewClusterOperatorHandler(operatorClient),
 		crWorkqueue:    workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "samplesconfig-changes"),
 		osSecWorkqueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "openshift-secret-changes"),
-		opSecWorkqueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "openshift-config-namespace-secret-changes"),
+		ocSecWorkqueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "openshift-config-namespace-secret-changes"),
+		opSecWorkqueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "openshift-cluster-samples-operator-secret-changes"),
```

Comment:
if i understand this you're watching the TLS secret in etcd so you can restart the metrics server when it changes.

but you don't need to do that..the secret is mounted to the pod, when it changes the file on disk will change.  So you can just watch the on-disk file for changes and restart when it changes.  Again there's a pattern for this that @ricardomaraschini recently implemented.

Reply:
yep per above that sounds better ... will look for it

Reply:
fyi the change was https://github.com/openshift/cluster-image-registry-operator/pull/342/commits/ffbb403ef87ede359e941891a28378f80dfe105e

Reply:
part of next push